x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/components/analytics_list/use_table_settings.ts

```diff
@@ -29,19 +30,22 @@ interface AnalyticsBasicTableSettings {
 }
 
 interface UseTableSettingsReturnValue {
-  onTableChange: EuiBasicTableProps<DataFrameAnalyticsListRow>['onChange'];
-  pageOfItems: DataFrameAnalyticsListRow[];
+  onTableChange: EuiBasicTableProps<DataFrameAnalyticsListRow | ModelItem>['onChange'];
+  pageOfItems: DataFrameAnalyticsListRow[] | ModelItem[];
   pagination: EuiBasicTableProps<DataFrameAnalyticsListRow>['pagination'];
   sorting: EuiTableSortingType<any>;
 }
 
-export function useTableSettings(items: DataFrameAnalyticsListRow[]): UseTableSettingsReturnValue {
+export function useTableSettings(
+  sortByField: string,
+  items: DataFrameAnalyticsListRow[] | ModelItem[]
```

Comment:
It would be better to use generics instead. I see
```typescript
const getPageOfItems = (
    list: any[],
```
in this hook, so you actually ignore the type entirely afterward. Generics should help with type safety.

Reply:
Updated types 👍  e132747913497159f245f33bca035eb9e2893cac